mods/map/init.lua

```diff
@@ -0,0 +1,84 @@
+-- Mod global namespace
+
+map = {}
+
+
+-- Cache creative mode setting
+
+local creative_mode_cache = minetest.settings:get_bool("creative_mode")
+
+
+-- Global function to allow overriding
+
+function map.enable(player)
+	local creative_enabled =
+		(creative and creative.is_enabled_for(player:get_player_name())) or
+		creative_mode_cache
+
+	if creative_enabled then
+		player:hud_set_flags({minimap = true})
+		player:hud_set_flags({minimap_radar = true})
+	else -- Survival
+		if player:get_inventory():contains_item("main", "map:mapping_kit") then
+			player:hud_set_flags({minimap = true})
+		else
+			player:hud_set_flags({minimap = false})
+		end
+		player:hud_set_flags({minimap_radar = false})
+	end
+end
+
+
+-- Check for map on joinplayer
+
+minetest.register_on_joinplayer(function(player)
+	map.enable(player)
+end)
+
+
+-- Regular check for map
+
+local function regular_check()
+	for _, player in ipairs(minetest.get_connected_players()) do
+		map.enable(player)
+	end
+	minetest.after(7, regular_check)
+end
+
+minetest.after(7, regular_check)
+
+
+-- Items
+
+minetest.register_craftitem("map:mapping_kit", {
+	description = "Mapping Kit",
+	inventory_image = "map_mapping_kit.png",
+	stack_max = 1,
+	groups = {flammable = 3},
+	-- To instantly enable minimap
+	on_use = function(itemstack, user, pointed_thing)
+		map.enable(user)
+		return nil
```

Comment:
Superfluous return value.

Reply:
Why? Docs say "Function must return either nil if no item shall be removed from
inventory, ..."

Reply:
Oh because `return nil` is the same as leaving the line out?

Reply:
Ok well it seems to work without that line.

Reply:
Yes, paramat.